assets/js/Containers/FrevaGPT/index.js

```diff
@@ -260,97 +278,126 @@ class FrevaGPT extends React.Component {
+                      >
+                        {truncate(element)}
+                      </Button>
+                    </OverlayTrigger>
+                  </div>
+                );
+              })}
+            </Row>
+          ) : null}
+
+          <ChatBlock />
+
+          {this.state.loading ? (
+            <Row className="mb-3">
+              <Col md={1}>
+                <Spinner />
+              </Col>
+            </Row>
+          ) : null}
+
+          <Row>
+            <Col md={12}>
+              <InputGroup className="mb-2 pb-2">
+                <FormControl
+                  type="text"
+                  value={this.state.userInput}
+                  onChange={this.handleUserInput}
+                  onKeyDown={this.handleKeyDown}
+                  placeholder="Ask a question"
+                />
+                {this.state.loading ? (
+                  <Button variant="outline-danger" onClick={this.handleStop}>
+                    <i className="bi bi-stop-fill"></i>
+                  </Button>
+                ) : (
+                  <Button variant="outline-success" onClick={this.handleSubmit}>
+                    <i className="bi bi-play-fill"></i>
+                  </Button>
+                )}
+              </InputGroup>
+            </Col>
+          </Row>
+        </Col>
+      </>
+    );
+  }
+
+  renderBotHeader() {
+    return (
+      <div className="d-flex justify-content-between mb-2">
+        <Form.Select
+          value={this.botModel}
+          onChange={(e) => {
+            this.setState({ botModel: e.target.value });
+          }}
+          className="me-1"
+          placeholder="Model"
+          hidden={this.state.hideBotModelList}
+        >
+          {this.state.botModelList.map((x) => {
+            return <option key={x}>{x}</option>;
+          })}
+        </Form.Select>
+        <Button onClick={this.createNewChat} variant="info">
+          NewChat
+        </Button>
+      </div>
+    );
+  }
+
   render() {
     return (
       <Container>
         <Row>
           <div className="d-flex justify-content-between">
             <h2 onClick={this.toggleBotSelect}>FrevaGPT</h2>
 
-            <div className="d-flex justify-content-between mb-2">
-              <Form.Select
-                value={this.botModel}
-                onChange={(e) => {
-                  this.setState({ botModel: e.target.value });
-                }}
-                className="me-1"
-                placeholder="Model"
-                hidden={this.state.hideBotModelList}
-              >
-                {this.state.botModelList.map((x) => {
-                  return <option key={x}>{x}</option>;
-                })}
-              </Form.Select>
-              <Button onClick={this.createNewChat} variant="info">
-                NewChat
-              </Button>
-            </div>
+            {this.state.botOkay === undefined
```

Comment:
since `undefined` is falsy as well, you can reduce this statement to

```
{ 
  this.state.botOkay ?  this.renderBotHeader() : null
}
```

Other than that, LGTM. I will annoy you with more change requests on your FrevaGPT-Branch, as soon as you open it for reviews :stuck_out_tongue: